src/schemas/text-based-fields.ts

```diff
@@ -122,6 +119,40 @@ const generateValidationSchema = (elementType: EElementType) => {
                 })
             );
         }
+        case EElementType.NUMERIC: {
+            return yup.array().of(
+                yup.object().shape({
+                    validationType: yup
+                        .string()
+                        .required("Validation required."),
+                    validationRule: yup.string().when("validationType", {
+                        is: ELEMENT_VALIDATION_TYPES["Text field"][
+                            EElementType.NUMERIC
+                        ].validationTypes[0],
```

Comment:
instead of checking based on array position, check for the exact enum
this prevents the logic from breaking if the array values change

```suggestion
                        is: EValidationType.EMAIL_DOMAIN,
```